### ISSUE TITLE
UnicodeEncodeError: 'ascii' codec can't encode character u'\u2026' in position 382: ordinal not in range(128)

### ISSUE THREAD

**Issue by okurz** [okurz/openqa_review#85]
## Observation

```
Traceback (most recent call last):
  File "/usr/bin/openqa-review", line 9, in <module>
    load_entry_point('openqa-review==0.0.0', 'console_scripts', 'openqa-review')()
  File "/usr/lib/python2.7/site-packages/openqa_review/openqa_review.py", line 1245, in main
    print(report)
  File "/usr/lib/python2.7/site-packages/openqa_review/openqa_review.py", line 1141, in __str__
    report_str += '# %s\n\n%s\n---\n' % (k, v)
  File "/usr/lib/python2.7/site-packages/openqa_review/openqa_review.py", line 958, in __str__
    'arch_report': '\n---\n'.join(map(str, self.reports.values()))
  File "/usr/lib/python2.7/site-packages/openqa_review/openqa_review.py", line 892, in __str__
    'existing_product_issues': issue_listing('**Existing Product bugs:**', self.issues['existing']['product'], self.args.show_empty),
  File "/usr/lib/python2.7/site-packages/openqa_review/openqa_review.py", line 342, in issue_listing
    return '\n' + header + '\n\n' + ''.join(map(str, issues)) + '\n'
UnicodeEncodeError: 'ascii' codec can't encode character u'\u2026' in position 382: ordinal not in range(128)
```

found in email report from daily dashboard report generation. I assume a problem introduced by any person using a label with non-ascii characters.

## Reproducible

* Happens sometimes but not every day in the daily jobs

## Further details
"daily dashboard report generation" is mainly https://github.com/okurz/openqa_review/blob/master/bin/openqa-review-daily-email triggered on a daily base from a cron job on lord.arch.suse.de for both o3 as well as osd.

**Comment by Zaoliang** [comment]
I have also an issue to report here

--zluo@e13:~/Documents/openQA/openqa_review/openqa_review> openqa-review -vvv -J https://openqa.suse.de/group_overview/110 -b 642.3,635.1 --no-empty-sections --bugrefs -TTT --query-issue-status
DEBUG:openqa_review.openqa_review:args: Namespace(abbreviate_test_issues=False, against_reviewed=None, arch=None, bugrefs=True, builds='642.3,635.1', dry_run=False, exclude_job_groups=None, filter=None, host='https://openqa.opensuse.org', include_softfails=False, job_group_urls='https://openqa.suse.de/group_overview/110', job_groups=None, load=False, load_dir='.', min_days_unchanged=14, no_progress=False, output_state_results=False, query_issue_status=True, query_issue_status_help=False, reminder_comment_on_issues=False, report_links=False, running_threshold=0, save=False, save_dir='.', short_failure_str=False, show_empty=False, verbose=4, verbose_test=4)
INFO:openqa_review.openqa_review:Acting on specified job group URL(s): https://openqa.suse.de/group_overview/110
INFO:openqa_review.openqa_review:Processing 'https://openqa.suse.de/group_overview/110'
WARNING:openqa_review.browser:Request to https://openqa.suse.de/group_overview/110.json was not successful after multiple retries, giving up
Traceback (most recent call last):
  File "/usr/bin/openqa-review", line 9, in <module>
    load_entry_point('openqa-review==0.0.0', 'console_scripts', 'openqa-review')()
  File "/usr/lib/python2.7/site-packages/openqa_review/openqa_review.py", line 1220, in main
    report = generate_report(args)
  File "/usr/lib/python2.7/site-packages/openqa_review/openqa_review.py", line 1156, in generate_report
    return Report(browser, args, root_url, job_groups)
  File "/usr/lib/python2.7/site-packages/openqa_review/openqa_review.py", line 1109, in __init__
    self.report[k] = self._one_report(v)
  File "/usr/lib/python2.7/site-packages/openqa_review/openqa_review.py", line 1117, in _one_report
    return ProductReport(self.browser, job_group_url, self.root_url, self.args)
  File "/usr/lib/python2.7/site-packages/openqa_review/openqa_review.py", line 894, in __init__
    current_url, previous_url = get_build_urls_to_compare(browser, job_group_url, args.builds, args.against_reviewed, args.running_threshold)
  File "/usr/lib/python2.7/site-packages/openqa_review/openqa_review.py", line 423, in get_build_urls_to_compare
    job_group = browser.get_json('%s.json' % job_group_url)
  File "/usr/lib/python2.7/site-packages/openqa_review/browser.py", line 75, in get_json
    return self.get_page(url, as_json=True, cache=cache)
  File "/usr/lib/python2.7/site-packages/openqa_review/browser.py", line 102, in get_page
    content = self._get(absolute_url, as_json=as_json)
  File "/usr/lib/python2.7/site-packages/openqa_review/browser.py", line 128, in _get
    raise DownloadError(msg)

**Comment by okurz** [comment]
That's not related at all. "WARNING:openqa_review.browser:Request to https://openqa.suse.de/group_overview/110.json was not successful after multiple retries, giving up" is the relevant part. Looks like you can simply not access that address from whatever machine you called this.

**Comment by nicksinger** [comment]
@Zaoliang you may have hit the misleading error message which gets clarified in #81 . A workaround is mentioned in #86

**Comment by nicksinger** [comment]
FYI: u'\u2026' is the [ellipsis char "…"](https://www.fileformat.info/info/unicode/char/2026/index.htm). I just wonder _what_ is enforcing the ASCII conversion here - IMHO that's the real bug here.
**Could** be related to the implicit casting to string here:

```
File "/usr/lib/python2.7/site-packages/openqa_review/openqa_review.py", line 342, in issue_listing
   return '\n' + header + '\n\n' + ''.join(map(str, issues)) + '\n'
```

**Comment by OleksandrOrlov** [comment]
@okurz could you please provide more detailed steps to reproduce? Also, what is the 'daily dashboard report generation' and where can I find it? Thank you.

**Comment by okurz** [comment]
@OleksandrOrlov Updated original description to provide more information about what is the "daily dashboard report generation".

For reproduction: Seems to happen not every time but was reported by multiple daily jobs. Maybe the easiest way to reproduce is to simulate some bug labels which have non-ASCII characters included in the subject.

**Comment by OleksandrOrlov** [comment]
Currently, I'm not able to reproduce the issue with the labels, that contain non-ascii characters.

I've subscribed to openqa daily reports and I'll keep track on this issue. If it will occur again, I'll gather more details and will try to fix it.

**Comment by okurz** [comment]
Yeah, the mailing list you subscribed to will not show you errors, only the generated reports in case of success

**Comment by OleksandrOrlov** [comment]
Finally, I've found what caused the issue. Today it was 
[poo#38648](https://progress.opensuse.org/issues/38648 "[sle12sp4] test fails in zypper_patch -- send “0” key Infinitely") - Please note that "0" is wrapped with double quotes (curly), that are not an ASCII characters. 

In previous occurrence it was ellipsis char.

Working on fix and tests for the issue.

**Comment by okurz** [comment]
Fixed with 4189852ebfd83252beec607ed5bc34ed2fc0f4ec
